component/mcsdadmin/component.go

```diff
@@ -0,0 +1,205 @@
+package mcsdadmin
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/nuts-foundation/nuts-knooppunt/component"
+	"github.com/rs/zerolog/log"
+	"html/template"
+	"net/http"
+)
+
+var _ component.Lifecycle = (*Component)(nil)
+
+type Component struct {
+}
+
+func New() *Component {
+	return &Component{}
+}
+
+func (c Component) Start() error {
+	// Nothing to do
+	return nil
+}
+
+func (c Component) Stop(ctx context.Context) error {
+	// Nothing to do
+	return nil
+}
+
+// Template rendering
+
+const templateFolder = "./component/mcsdadmin/templates/"
+
+func RenderWithBase(w http.ResponseWriter, name string, data any) {
+	files := []string{
+		templateFolder + "base.html",
+		templateFolder + name,
+	}
+
+	ts, err := template.ParseFiles(files...)
```

Comment:
this won't work when you've built the packaged (e.g. in a Docker container), because then you won't have the template files. We need to embed these assets using `go:embed`, which can then be used as filesystem when reading the template files. Good starting point: https://pkg.go.dev/embed

That said, during development it's useful to have this "live reload" function as you now have, so maybe we'd have it configurable (or, if there a "templates" directory, use that instead of the embedded fs)

Reply:
Pretty cool feature these file system embeds. Using them now.